depmapomics/copynumbers.py

```diff
@@ -682,50 +873,66 @@ def postProcess(
     )
     mybiomart["Chromosome"] = mybiomart["Chromosome"].astype(str)
     mybiomart = mybiomart.sort_values(by=["Chromosome", "start", "end"])
-    mybiomart = mybiomart[mybiomart["Chromosome"].isin(set(segments["Chromosome"]))]
-    mybiomart = mybiomart.drop_duplicates("ensembl_gene_id", keep="first")
-    # drop Ychrom if > maxYchrom
-    ychrom = segments[segments.Chromosome.str.contains("Y")]
-    countYdrop = [
-        i
-        for i in set(ychrom[constants.SAMPLEID])
-        if len(ychrom[ychrom[constants.SAMPLEID] == i]) > maxYchrom
+    mybiomart = mybiomart[
+        mybiomart["Chromosome"].isin([str(x) for x in range(1, 23)] + ["X", "Y"])
     ]
-    segments = segments[
-        ~(
-            (segments[constants.SAMPLEID].isin(countYdrop))
-            & (segments.Chromosome == "Y")
-        )
-    ]
-    genecn = mut.toGeneMatrix(
-        mut.manageGapsInSegments(segments),
-        mybiomart,
-        value_colname="SegmentMean",
-        gene_names_col="ensembl_gene_id",
-    )
-    # validation step
-    print("summary of the gene cn data:")
-    print(genecn.values.min(), genecn.values.mean(), genecn.values.max())
-    failed = mut.checkAmountOfSegments(segments, thresh=segmentsthresh)
-
-    print("failed our QC")
-    print(failed)
-
-    if source_rename:
-        segments = segments.replace({"Source": source_rename})
-    if save_output:
-        h.listToFile(failed, save_output + "failed.txt")
-    # subsetting
-    segments = segments[
-        ~segments[constants.SAMPLEID].isin((set(failed) | set(todrop)) - set(priority))
-    ].reset_index(drop=True)
-    genecn = genecn[~genecn.index.isin((set(failed) | set(todrop)) - set(priority))]
+    mybiomart = mybiomart.drop_duplicates("ensembl_gene_id", keep="first")
 
-    # saving
-    print("saving files")
-    segments.to_csv(save_output + "segments_all.csv", index=False)
-    genecn.to_csv(save_output + "genecn_all.csv")
-    print("done")
+    if run_gatk_relative:
+        segments = loadFromGATKAggregation(
+            refworkspace,
```

Comment:
@dpmccabe are results from HMM accessible in the DepMap_WGS_CN workspace or should we pass in a different workspace here?

Reply:
Yes, `cnv_cn_by_gene_weighted_mean` and `cnv_segments` are the HMM CNV columns I've copied over, but isn't this code here for the pre-HMM aggregation? The `else` below is the new logic branch.

Reply:
Oops, I meant to comment this on l931, but this answers my question!